desc/plotting.py

```diff
@@ -933,282 +1009,327 @@ def plot_comparison(
 
-    return out.reshape((grid.M, grid.L, grid.N), order="F")
+    ax.set_xlabel(_axis_labels_rtz[0])
+    ax.set_ylabel(r"current $(A)$")
+    fig.legend(loc="center right")
+    fig.set_tight_layout(True)
+    return fig, ax
 
 
-def _format_name(name):
-    """Parse name string into dictionary.
+def plot_boozer_modes(
+    eq, log=True, B0=True, num_modes=-1, L=20, rho=None, ax=None, **kwargs
+):
+    """Plot Fourier harmonics of :math:`|B|` in Boozer coordinates.
 
     Parameters
     ----------
-    name : str
+    eq : Equilibrium
+        Object from which to plot.
+    log : bool, optional
+        Whether to use a log scale.
+    B0 : bool, optional
+        Whether to include the m=n=0 mode.
+    num_modes : int, optional
+        How many modes to include. Default (-1) is all.
+    L : int, optional
+        Number of flux surfaces to evaluate at. Only used if rho=None.
+    rho : ndarray, optional
+        Radial coordinates of the flux surfaces to evaluate at.
+    ax : matplotlib AxesSubplot, optional
+        Axis to plot on.
 
     Returns
     -------
-    parsed name : dict
+    fig : matplotlib.figure.Figure
+        Figure being plotted to.
+    ax : matplotlib.axes.Axes or ndarray of Axes
+        Axes being plotted to.
 
     """
-    name_dict = {"base": "", "sups": "", "subs": "", "power": "", "d": "", "units": ""}
-    if "**" in name:
-        parsename, power = name.split("**")
-        if "_" in power or "^" in power:
-            raise SyntaxError(
-                "Power operands must come after components and derivatives."
-            )
+    if rho is None:
+        rho = np.linspace(1, 0, num=L, endpoint=False)
+    ds = []
+    B_mn = np.array([[]])
+    linestyle = kwargs.get("linestyle", "-")
+    for i, r in enumerate(rho):
+        grid = LinearGrid(M=6 * eq.M + 1, N=6 * eq.N + 1, NFP=eq.NFP, rho=r)
+        data = eq.compute("|B|_mn", grid)
+        ds.append(data)
+        b_mn = np.atleast_2d(data["|B|_mn"])
+        B_mn = np.vstack((B_mn, b_mn)) if B_mn.size else b_mn
+    idx = np.argsort(np.abs(B_mn[0, :]))
+    if num_modes == -1:
+        idx = idx[-1::-1]
     else:
-        power = ""
-        parsename = name
-    name_dict["power"] += power
-    if "_" in parsename:
-        split = parsename.split("_")
-        if len(split) == 3:
-            name_dict["base"] += split[0]
-            name_dict["subs"] += split[1]
-            name_dict["d"] += split[2]
-        elif "^" in split[0]:
-            name_dict["base"], name_dict["sups"] = split[0].split("^")
-            name_dict["d"] = split[1]
-        elif len(split) == 2:
-            name_dict["base"], other = split
-            if other in ["rho", "theta", "zeta", "beta", "TP", "FF"]:
-                name_dict["subs"] = other
-            else:
-                name_dict["d"] = other
+        idx = idx[-1 : -num_modes - 1 : -1]
+    B_mn = B_mn[:, idx]
+    modes = data["B modes"][idx, :]
```

Comment:
This line throws an error, not sure what B modes is supposed to be

Reply:
This is resolved now.